### ISSUE TITLE
Create CI/CD pipeline for project

### ISSUE THREAD

**Issue by terrameijar** [mpumzee/ChitChat-front#10]
We want to follow a CI/CD approach to building and deploying this project.

## Continuous Integration

We'll use GitHub actions for CI to automatically run tests, lint checks and builds on new code that gets pushed to the `develop` and `main` branches.

## Continuous Deployment

We want to automatically deploy code to production, and we'll use a combination of Ansible and GitHub webhooks.

Create a CI/CD pipeline that will do all the above.

**Comment by ChikangaTakudzwa** [comment]
are these to be done now??, i like working on DevOps stuff.

**Comment by terrameijar** [comment]
@ChikangaTakudzwa yes. This task is in our [Initial Go-Live milestone](https://github.com/PythonBulawayo/ChitChat/milestone/1) so it's one of our highest priorities. We don't currently have tests but we could start with a CI pipeline that does lint and format checks in GitHub Actions. Once we get our hosting issues sorted, we can work on continuous deployment.

The way I see it, the flow of steps will look something like this:

For the CI stage:

- [ ] Add code linters to the project such as [flake8](https://flake8.pycqa.org/en/latest/) for Python and [ESLint](https://eslint.org/) or [Prettier](https://prettier.io/) the FrontEnd
- [ ] Create GitHub Actions for linting both Front End and Back End code

For the CD stage:
- [ ] Create an Ansible playbook that will checkout the code to the deployment server and run it 
- [ ] Add a GitHub Action that will SSH into our deployment server and run the ansible playbook


 Would you be happy to work on any of these tasks? I'm happy to help you with any part.

**Comment by Thabhelo** [comment]
I'd love to contribute to:

Adding code linters to the project such as [flake8](https://flake8.pycqa.org/en/latest/) for Python and [ESLint](https://eslint.org/) or [Prettier](https://prettier.io/) the FrontEnd

**Comment by terrameijar** [comment]
@Thabhelo, wonderful, thank you for volunteering to help. This is a straightforward task. Please setup fork and clone the project to your environment and let me know here once you're done so I can let you know what next steps to take.

**Comment by ChikangaTakudzwa** [comment]
@terrameijar Yes i can handle them, so i guess i will be working with @Thabhelo ,, @Thabhelo can you give me a heads up on what you have done or what you are working on first so that we move together.

**Comment by ChikangaTakudzwa** [comment]
i think i will work on these tomorrow

**Comment by terrameijar** [comment]
Let's split the work, @ChikangaTakudzwa please add flake8 to the project and @Thabhelo, add ESLint. We want to have flake8 and ESlint dependencies added to the correct files for each tier(requirements.txt for the backend and package.json in the frontend.

Let's start on these tasks before implementing GitHub Actions.

**Comment by terrameijar** [comment]
@Thabhelo I have created an issue for you https://github.com/PythonBulawayo/ChitChat/issues/31 to track progress on adding ESLint.

**Comment by Nolulamo** [comment]
I will like to contribute to #27

**Comment by terrameijar** [comment]
@ChikangaTakudzwa I have created a separate issue to track the flake8 task here: https://github.com/PythonBulawayo/ChitChat/issues/32. Once you're done, please comment on it and let me know.

**Comment by ChikangaTakudzwa** [comment]
okay i am on it
